tools/mcsema_disass/ida/get_cfg.py

```diff
@@ -1324,6 +1324,43 @@ def identify_program_entrypoints(func_eas):
   DEBUG_POP()
   return exported_funcs, exported_vars
 
+def find_main_in_ELF_file():
+  """Tries to automatically find the `main` function if we haven't found it
+  yet. IDA recognizes the pattern of `_start` calling `__libc_start_main` in
+  ELF binaries, where one of the parameters is the `main` function. IDA will
+  helpfully comment it as such."""
+
+  start_ea = idc.LocByName("_start")
+  if is_invalid_ea(start_ea):
+    start_ea = idc.LocByName("start")
+    if is_invalid_ea(start_ea):
+      return idc.BADADDR
+
+  for begin_ea, end_ea in idautils.Chunks(start_ea):
+    for inst_ea in Heads(begin_ea, end_ea):
+      comment = idc.GetCommentEx(inst_ea, 0)
+      if comment and "main" in comment:
+        for main_ea in xrefs_from(inst_ea):
+          if not is_code(main_ea):
+            continue
+
+          # Sometimes the `main` function isn't identified as code. This comes
+          # up when there are some alignment bytes in front of `main`.
+          try_mark_as_code(main_ea)
+          if is_code_by_flags(main_ea):
+            try_mark_as_function(main_ea)
+
+          main = idaapi.get_func(main_ea)
+          if not main:
+            continue
+
+          if main and main.startEA == main_ea:
+            set_symbol_name(main_ea, "main")
+            DEBUG("Found main at {:x}".format(main_ea))
+            return main_ea
+
+  return idc.BADADDR
```

Comment:
Is it worth having a `DEBUG("Main function not found")` here?

Reply:
It actually comes up as an `ERROR:` down the line, saying that the `--entrypoint` couldn't be found.